### PR TITLE
Fixes #6832 - Don't show location/organization on host edit page if host...

### PR DESCRIPTION
**Pull request by tbrisker**
... has none

### CODE REVIEW

File: app/views/hosts/_form.html.erb [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
         <% if show_organization_tab?  %>
           <%= select_f f, :organization_id, Organization.my_organizations, :id, :to_label,
-            { :include_blank => !@host.managed?, :selected => @organization.try(:id)},
+            { :include_blank => !@host.managed? || @host.organization_id.nil?, :selected => @host.organization_id },
```
Comment (ohadlevy) [inline app/views/hosts/_form.html.erb]: i think we want to default to @organization, to ensure that the current  org context is preselected?

File: app/views/hosts/_form.html.erb [hunk 2]
```diff
@@ -41,7 +41,7 @@
 
         <% if show_location_tab? %>
           <%= select_f f, :location_id, Location.my_locations, :id, :to_label,
-            { :include_blank => !@host.managed?, :selected => @location.try(:id) },
+            { :include_blank => !@host.managed? || @host.location_id.nil?, :selected => @host.location_id },
```
Comment (ohadlevy) [inline app/views/hosts/_form.html.erb]: ditto.

any useful tests for this?

File: app/views/hosts/_form.html.erb (hunk 1 above)
Comment (tbrisker) [inline app/views/hosts/_form.html.erb]: The problem is defaulting to `@organization` leads to hosts with no organization showing the current (or first if context is any) on edit view. Maybe only default to current on creation view when a context is selected (i.e. not any)
btw, why is changing org/loc disabled after host creation?

Comment (ohadlevy) [inline app/views/hosts/_form.html.erb]: About creation, probably the right way forward.

About why, is that everything is pretty much scoped to org/loc and would
usually mean you would lose all of your form data, in the past, it was a
major effort to implement it ui wize, so we decided against it. can be
revisited of course.

**Comment by dLobatog** [comment]
Merged as 65292474dcf224acf68d3b23741548553bf34629, thanks @tbrisker !

File: app/controllers/hosts_controller.rb [hunk 3]
```diff
@@ -625,14 +625,8 @@ def taxonomy_scope
     @organization ||= Organization.find_by_id(params[:organization_id]) if params[:organization_id]
     @location     ||= Location.find_by_id(params[:location_id])         if params[:location_id]
 
-    if SETTINGS[:organizations_enabled]
-      @organization ||= Organization.current
-      @organization ||= Organization.my_organizations.first
```
Comment (ohadlevy) [inline app/controllers/hosts_controller.rb:630]: what happens when you have more than one org and you are in any context?

Comment (dLobatog) [inline app/controllers/hosts_controller.rb:630]: Nothing special I think, `@organization` and `@location` would default to blank.

Comment (tbrisker) [inline app/controllers/hosts_controller.rb:630]: They would default to `nil`. 
I think anyhow the correct way to handle the "Any" context is not by defaulting to the first one.

**Pull request closed without merging**